cmd/internal/scdiff/app/stats.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+//nolint:gochecknoinits // common for cobra apps
+func init() {
+	rootCmd.AddCommand(statsCmd)
+	statsCmd.PersistentFlags().StringVarP(&statsCheck, "check", "c", "", "Analyze breakdown of a single check")
+}
+
+// 1 MiB size limit for individual results. This currently works,
+// but bufio.Scanner always has a limit, may need to change approach in the future.
+const maxResultSize = 1024 * 1024
+
+var (
+	statsCheck string
+	statsCmd   = &cobra.Command{
+		Use:   "stats [flags] FILE",
+		Short: "Summarize stats for a golden file",
+		Long:  `Summarize stats for a golden file`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errMissingInputFiles
+			}
+			f1, err := os.Open(args[0])
```

Comment:
Can you make this a required `StringVar` with cobra?

Reply:
Hmm, I was taking a `diff` approach, where the file is expected as an argument but isn't a flag.
It seems like there are better ways of doing what I was trying to do:
https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#positional-and-custom-arguments

Reply:
applied this change using the cobra arg validation, if you could please take another look